### PR TITLE
Use a heuristic to set ignore-system-power for all Lenovo batteries

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: src/fu-engine.c [hunk 1]
```diff
@@ -4512,6 +4512,23 @@ fu_engine_md_refresh_device_name_category(FuEngine *self, FuDevice *device, XbNo
 		fu_device_remove_internal_flag(device,
 					       FU_DEVICE_INTERNAL_FLAG_MD_SET_NAME_CATEGORY);
 	}
+
+	/* using GUIDs to set internal flag ignore-system-power doesn't scale well */
+	if (fu_device_has_vendor_id(device, "DMI:LENOVO")) {
```
Comment (superm1) [inline src/fu-engine.c:4517]: why restrict this to Lenovo?

File: src/fu-engine.c [hunk 2]
```diff
@@ -4512,6 +4512,23 @@ fu_engine_md_refresh_device_name_category(FuEngine *self, FuDevice *device, XbNo
 		fu_device_remove_internal_flag(device,
 					       FU_DEVICE_INTERNAL_FLAG_MD_SET_NAME_CATEGORY);
 	}
+
+	/* using GUIDs to set internal flag ignore-system-power doesn't scale well */
```
Comment (superm1) [inline src/fu-engine.c:4516]: I think this comment should be adjusted

File: src/fu-engine.c (hunk 1 above)
Comment (hughsie) [reply src/fu-engine.c:4517]: I guess what i'm trying to do is restrict it to laptop batteries, but we don't have enough metadata to know the difference between a battery in a laptop and a battery in a wireless mouse for instance.

Comment (superm1) [reply src/fu-engine.c:4517]: you could just only do it with the uefi-capsule plugin instead then perhaps?

File: src/fu-engine.c (hunk 2 above)
Comment (hughsie) [reply src/fu-engine.c]: Fixed.

File: src/fu-engine.c (hunk 1 above)
Comment (hughsie) [reply src/fu-engine.c]: Yup, that's a good heuristic I guess. Fixed.

**Comment by hughsie** [comment]
> I don't recall if we have documentation for the X-Foo stuff in tree

LVFS yes, fwupd no. Thanks.

**Pull request merged**